app/models/call_flow_logic/ews_registration.rb

```diff
@@ -367,15 +367,24 @@ def persist_commune
     def update_contact
       contact = phone_call.contact
       commune = Pumi::Commune.find_by_id(phone_call_metadata(:commune_code))
+
+      contact.addresses.find_or_create_by!(
+        iso_region_code: commune.province_id,
```

Comment:
use this one instead

![IMG_0900](https://github.com/user-attachments/assets/bdb8fb33-2186-4399-8148-a1ae0dd5ffec)

Reply:
Ah, got it.